src/components/SelectionListSingle/ListHeader.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import {View} from 'react-native';
+import Checkbox from '@components/Checkbox';
+import {PressableWithFeedback} from '@components/Pressable';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+import type {ListItem} from './types';
+
+type HeaderProps<TItem extends ListItem> = {
+    dataDetails: DataDetailsType<TItem>;
+    headerMessage?: string;
+    customListHeader?: React.ReactNode;
+    canSelectMultiple: boolean;
+    onSelectAll?: () => void;
+    shouldPreventDefaultFocusOnSelectRow?: boolean;
+};
+
+type DataDetailsType<TItem extends ListItem> = {
+    allOptions: TItem[];
+    selectedOptions: TItem[];
+    allSelected: boolean;
+    someSelected: boolean;
+    disabledIndexes: number[];
+    disabledArrowKeyIndexes: number[];
+};
+
+function SelectionListHeader<TItem extends ListItem>({
+    dataDetails,
+    headerMessage,
+    customListHeader,
+    canSelectMultiple,
+    onSelectAll,
+    shouldPreventDefaultFocusOnSelectRow,
+}: HeaderProps<TItem>) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    if (headerMessage) {
+        return null;
+    }
```

Comment:
There is something odd about this code 😄 Why is this passed as a prop and we return null if there is a message? It seems like it should be the opposite